projects/dex-ui/src/wells/useBeanstalkSiloWhitelist.ts

```diff
@@ -1,24 +1,36 @@
-import { useMemo } from "react";
+import { useCallback } from "react";
 import { Well } from "@beanstalk/sdk/Wells";
+import { BEANETH_MULTIPUMP_ADDRESS } from "src/utils/addresses";
+import useSdk from "src/utils/sdk/useSdk";
 
-const WHITELIST_MAP = {
-  /// BEANWETHCP2w (BEANETH LP)
-  "0xbea0e11282e2bb5893bece110cf199501e872bad": {
-    address: "0xBEA0e11282e2bB5893bEcE110cF199501e872bAd",
-    lpTokenAddress: "0xbea0e11282e2bb5893bece110cf199501e872bad"
-  }
-};
-
-/// set of wells that are whitelisted for the Beanstalk silo
 export const useBeanstalkSiloWhitelist = () => {
-  const whitelistedAddresses = useMemo(() => Object.keys(WHITELIST_MAP), []);
+  const sdk = useSdk();
+
+  const getIsWhitelisted = useCallback(
+    (well: Well | undefined) => {
+      if (!well?.lpToken) return false;
+      const token = sdk.tokens.findByAddress(well.lpToken.address);
+      return Boolean(token && sdk.tokens.siloWhitelist.has(token));
+    },
+    [sdk.tokens]
+  );
 
-  const getIsWhitelisted = (well: Well | undefined) => {
-    if (!well) return false;
-    const wellAddress = well.address.toLowerCase();
+  const getSeedsWithWell = useCallback(
+    (well: Well | undefined) => {
+      if (!well?.lpToken) return undefined;
+      return sdk.tokens.findByAddress(well.lpToken.address)?.getSeeds();
```

Comment:
In order for .getSeeds() to work, you MUST call sdk.refresh()

See how we do it in the UI:
https://github.com/AgrarianAlliance/Beanstalk/pull/45/files#diff-c46de2fb1f0f47ec5cdb8a6c77a955c9ec346362f812fb70e4a38ee0549645e1R71

Note, this requires BIP39, which hasn't been deployed yet so this means we cannot merge this PR until after.

Happy to chat about this if you have questions. You don't need to set the seed for each token like I did in the UI PR below the line commented, that was only to update the UI's tokens... the SDK will update its tokens on refresh().